invenio_app_ils/circulation/api.py

```diff
@@ -77,6 +92,12 @@ def request_loan(
         patron_pid=patron_pid, document_pid=document_pid
     ):
         raise PatronHasRequestOnDocumentError(patron_pid, document_pid)
+
+    if patron_has_active_loan_on_document(
```

Comment:
what about if we avoid 2 searches and we use the check above by checking also for this extra state?
You will be able to throw a different error based on the result eventually.

Can you also please add a test? It is important that we test all these cases.